### ISSUE TITLE
Fix S2479  FN: Support utf-8 strings

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6291]
### Description

FN for S2479 (ControlCharacterInString).

Using C#11 utf8 strings, the rule is not raising an issue.

### Repro steps

```csharp
ReadOnlySpan<byte> Utf8NonCompliant = "test"u8; // FN
ReadOnlySpan<byte> Uft8NonCompliant2 = """test"""u8; // FN
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

**Comment by gregory-paidis-sonarsource** [comment]
After internal conversation with  @csaba-sagi-sonarsource , we decided to ignore verbatim and/or raw utf-8 string, since they have no escape mechanism.

Examples:

```csharp
            ReadOnlySpan<byte> Utf8Compliant = "test"u8; // Compliant
            ReadOnlySpan<byte> Utf8Noncompliant = "test"u8; // Noncompliant
            ReadOnlySpan<byte> Utf8VerbatimCompliant = @"test"u8; // Compliant, verbatim utf-8 string

            ReadOnlySpan<byte> Utf8CompliantRaw = """test"""u8; // Compliant, raw string
            ReadOnlySpan<byte> Utf8CompliantRawWithSpecialCharacter = """test"""u8; // Compliant, raw string

            ReadOnlySpan<byte> Utf8CompliantVerbatimRaw = @"""test"""u8; // Compliant, raw+verbatim string
            ReadOnlySpan<byte> Utf8CompliantVerbatimRawWithSpecialCharacter = @"""test"""u8; // Compliant, raw+verbatim string

```
